### ISSUE TITLE
false positive SyntaxError

### ISSUE THREAD

**Issue by newcodepusher** [ConsenSys/scribble#28]
I have a false positive syntax error in a dependent contract from openzeppelin
[MerkleProof.sol](https://github.com/OpenZeppelin/openzeppelin-contracts/blob/release-v3.3/contracts/cryptography/MerkleProof.sol)

```solidity
/**
 * @dev These functions deal with verification of Merkle trees (hash trees),
 */
library MerkleProofUpgradeable {
    /**
     * @dev Returns true if a `leaf` can be proved to be a part of a Merkle tree
     * defined by `root`. For this, a `proof` must be provided, containing
     * sibling hashes on the branch from the leaf to the root of the tree. Each
     * pair of leaves and each pair of pre-images are assumed to be sorted.
     */
    function verify(bytes32[] memory proof, bytes32 root, bytes32 leaf) internal pure returns (bool) {
        bytes32 computedHash = leaf;

        for (uint256 i = 0; i < proof.length; i++) {
            bytes32 proofElement = proof[i];

            if (computedHash <= proofElement) {
                // Hash(current computed hash + current element of the proof)
                computedHash = keccak256(abi.encodePacked(computedHash, proofElement));
            } else {
                // Hash(current element of the proof + current computed hash)
                computedHash = keccak256(abi.encodePacked(proofElement, computedHash));
            }
        }

        // Check if the computed hash (root) is equal to the provided root
        return computedHash == root;
    }
}
```

log:
```
$ scribble --arm ./MerkleProof.sol

./MerkleProof.sol:11:15 SyntaxError: Expected "(" or whitespace but "b" found.

In:


     * defined by `root`. For this, a `proof` must be provided, containing
     * sibling hashes on the branch from the leaf to the root of the tree. Each
     * pair of leaves and each pair of pre-images are assumed to be sorted.
     */
```

scribble veriosn 0.4.0

**Comment by blitz-1306** [comment]
Thanks for reporting this. It appears that the word `define` at the begining of the line docstring. We will investigate.

**Comment by cd1m0** [comment]
The issue here is in the way we try to extract candidate annotations for parsing from unstructured text. As you can guess we use a fairly simple regex:

```
const rx = /\s*(\*|\/\/\/)\s*(if_succeeds|if_updated|if_assigned|invariant|define/g;
```

Since `if_succeeds/if_pdated/if_assigned` aren't common words this worked fine. However `invariant` and `define` are problematic - whenever they are the first word on a comment line they will mismatch. To fix your specific case we can make the regex for `define` more precise like so:

```
const rx = /\s*(\*|\/\/\/)\s*(if_succeeds|if_updated|if_assigned|invariant|define\s*[a-zA-Z0-9_]*\s*\([^)]*\))/g;
```

`invariant` still remains a problem though, since it can be followed by any solidity expression. For now we may add an FAQ entry for `invariant` just to warn people. Eventually this will be resolved, as we are working with the Solidity compiler team to migrate annotations to natspec. For details check out this issue: https://github.com/ethereum/solidity/issues/10825

**Comment by cd1m0** [comment]
Fix will be part of 0.4.2 to be out today or tomorrow
